### PR TITLE
[TEE] Set a random key to avoid other node joining the cluster.

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
We now support that the HEAD node is in TEE, and we should prevent other nodes joining to this cluster, as a result, give a random password for head node is a simple approach, with a minor change.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
@zhouaihui @ian-huu  @miluoalbert @NKcqx

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -601,6 +601,13 @@ def start(
             stacklevel=2,
         )
 
+    if os.getenv("ENABLE_TEE") == "true":
```
Comment (NKcqx) [inline python/ray/scripts/scripts.py:604]: In PR #380 , I used environment variable `RAY_IN_TEE` ,  let's choose a unified name

**Review by NKcqx — COMMENTED** [review]
This is indeed a minimal solution !

File: python/ray/scripts/scripts.py (hunk 1 above)
Comment (jovany-wang) [reply python/ray/scripts/scripts.py:604]: Yes, I noticed that. I'm not sure what the best name is here. Maybe let's @ others CC

**Review by NKcqx — APPROVED** [review]

**Pull request merged**